concept/type/impl/RelationTypeImpl.java

```diff
@@ -167,22 +168,15 @@ public void unsetRelates(String roleLabel) {
     }
 
     @Override
-    public FunctionalIterator<RoleTypeImpl> getRelates() {
-        FunctionalIterator<RoleTypeImpl> roles = vertex.outs().edge(RELATES).to().map(v -> RoleTypeImpl.of(graphMgr, v));
-        if (isRoot()) {
-            return roles;
-        } else {
-            assert getSupertype() != null;
-            Set<RoleTypeImpl> overridden = new HashSet<>();
-            overriddenRoles().forEachRemaining(overridden::add);
-            return link(roles, getSupertype().asRelationType().getRelates().filter(role -> !overridden.contains(role)));
```

Comment:
TODO this behaviour is not exactly replicated in the new change! We should only filter the supertypes' relates as we do in the old code